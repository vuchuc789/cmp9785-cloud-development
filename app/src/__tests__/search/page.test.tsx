import SearchPage from '@/app/search/page';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { act } from 'react';

jest.mock('next/navigation', () => ({
  useRouter: () => ({ replace: () => {} }),
  useSearchParams: () => new URLSearchParams(),
  usePathname: () => ({}),
}));

jest.mock('../../contexts/auth', () => ({
  useAuth: jest.fn(() => ({ state: { accessToken: 'abc', isLoading: false } })),
}));

jest.mock('../../client/', () => ({
  searchMediaMediaSearchGet: jest.fn(() => ({ error: { detail: [{}] } })),
  MediaType: { IMAGE: 'image' },
  MediaLicense: [],
}));

describe('SearchPage', () => {
  it('renders profile page unchanged', async () => {
    let container;
    await act(async () => {
      const result = render(<SearchPage />);

      container = result.container;
    });

    expect(container).toMatchSnapshot();
  });
});
