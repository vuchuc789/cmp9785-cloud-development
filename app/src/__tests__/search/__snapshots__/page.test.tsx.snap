// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders profile page unchanged 1`] = `
<div>
  <div
    class="grow max-w-screen-xl mx-auto px-4 sm:px-6 lg:px-8 w-full py-6 md:py-10"
  >
    <div
      class="mb-8"
    >
      <h1
        class="scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl"
      >
        Discover Media
      </h1>
      <p
        class="mt-2 text-xl text-muted-foreground"
      >
        Search for images and audio all in one place.
      </p>
    </div>
    <form
      class="relative"
    >
      <div
        class="relative flex w-full max-w-full items-center"
      >
        <div
          class="relative w-full"
        >
          <svg
            class="lucide lucide-search absolute left-3 top-1/2 h-4 w-4 -translate-y-1/2 text-muted-foreground"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
          <input
            aria-describedby="undefined-form-item-description"
            aria-invalid="false"
            class="border-input file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground flex w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive pl-10 pr-32 h-12 focus-visible:ring-0"
            data-slot="form-control"
            id="undefined-form-item"
            name="q"
            placeholder="Search for images and audio..."
            type="search"
            value=""
          />
        </div>
        <div
          class="absolute right-1 top-1/2 -translate-y-1/2 flex items-center gap-1"
        >
          <button
            aria-controls="radix-:r0:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground size-9 h-10 w-10 cursor-pointer"
            data-slot="popover-trigger"
            data-state="closed"
            type="button"
          >
            <svg
              class="lucide lucide-clock h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <polyline
                points="12 6 12 12 16 14"
              />
            </svg>
            <span
              class="sr-only"
            >
              Search history
            </span>
          </button>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 px-4 py-2 has-[>svg]:px-3 h-10 cursor-pointer"
            data-slot="button"
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
    </form>
    <div
      class="flex flex-col gap-8 md:flex-row mt-8"
    >
      <div
        class="hidden md:block w-[240px] flex-shrink-0"
      >
        <div
          class="space-y-6"
        >
          <div>
            <h3
              class="mb-4 text-lg font-medium"
            >
              Filters
            </h3>
            <div
              data-orientation="vertical"
              data-slot="accordion"
            >
              <div
                class="border-b last:border-b-0"
                data-orientation="vertical"
                data-slot="accordion-item"
                data-state="open"
              >
                <h3
                  class="flex"
                  data-orientation="vertical"
                  data-state="open"
                >
                  <button
                    aria-controls="radix-:r2:"
                    aria-expanded="true"
                    class="focus-visible:border-ring focus-visible:ring-ring/50 flex flex-1 items-start justify-between gap-4 rounded-md py-4 text-left text-sm font-medium transition-all outline-none hover:underline focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180 cursor-pointer"
                    data-orientation="vertical"
                    data-radix-collection-item=""
                    data-slot="accordion-trigger"
                    data-state="open"
                    id="radix-:r1:"
                    type="button"
                  >
                    License
                    <svg
                      class="lucide lucide-chevron-down text-muted-foreground pointer-events-none size-4 shrink-0 translate-y-0.5 transition-transform duration-200"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                </h3>
                <div
                  aria-labelledby="radix-:r1:"
                  class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
                  data-orientation="vertical"
                  data-slot="accordion-content"
                  data-state="open"
                  id="radix-:r2:"
                  role="region"
                  style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
                >
                  <div
                    class="pt-0 pb-4"
                  >
                    <div
                      class="space-y-2"
                    >
                      <div
                        class="flex flex-wrap gap-2 mb-2"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mt-4 space-y-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 w-full cursor-pointer"
                data-slot="button"
              >
                Apply Filters
              </button>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border border-input bg-background shadow-xs hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 has-[>svg]:px-3 w-full cursor-pointer"
                data-slot="button"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:hidden"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border border-input bg-background shadow-xs hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 has-[>svg]:px-3 mb-4 w-full"
          data-slot="sheet-trigger"
          data-state="closed"
          type="button"
        >
          <svg
            class="lucide lucide-filter mr-2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
            />
          </svg>
          Filters
        </button>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col gap-2 w-full"
          data-orientation="horizontal"
          data-slot="tabs"
          dir="ltr"
        >
          <div
            aria-orientation="horizontal"
            class="text-muted-foreground inline-flex items-center mb-6 border-b rounded-none w-full h-auto p-0 bg-transparent justify-start"
            data-orientation="horizontal"
            data-slot="tabs-list"
            role="tablist"
            style="outline: none;"
            tabindex="0"
          >
            <button
              aria-controls="radix-:r6:-content-image"
              aria-selected="true"
              class="data-[state=active]:bg-background data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:bg-input/50 h-[calc(100%-1px)] flex-1 justify-center text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 flex items-center gap-2 px-6 py-3 rounded-none border-0 border-b-2 border-transparent cursor-pointer data-[state=active]:border-gray-900 dark:data-[state=active]:border-gray-100 bg-transparent data-[state=active]:shadow-xs"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-slot="tabs-trigger"
              data-state="active"
              id="radix-:r6:-trigger-image"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                class="lucide lucide-image h-5 w-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="18"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="3"
                />
                <circle
                  cx="9"
                  cy="9"
                  r="2"
                />
                <path
                  d="m21 15-3.086-3.086a2 2 0 0 0-2.828 0L6 21"
                />
              </svg>
              <span>
                Images
              </span>
            </button>
            <button
              aria-controls="radix-:r6:-content-audio"
              aria-selected="false"
              class="data-[state=active]:bg-background data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:bg-input/50 h-[calc(100%-1px)] flex-1 justify-center text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 flex items-center gap-2 px-6 py-3 rounded-none border-0 border-b-2 border-transparent cursor-pointer data-[state=active]:border-gray-900 dark:data-[state=active]:border-gray-100 bg-transparent data-[state=active]:shadow-xs"
              data-orientation="horizontal"
              data-radix-collection-item=""
              data-slot="tabs-trigger"
              data-state="inactive"
              id="radix-:r6:-trigger-audio"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                class="lucide lucide-music h-5 w-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 18V5l12-2v13"
                />
                <circle
                  cx="6"
                  cy="18"
                  r="3"
                />
                <circle
                  cx="18"
                  cy="16"
                  r="3"
                />
              </svg>
              <span>
                Audio
              </span>
            </button>
          </div>
          <div
            aria-labelledby="radix-:r6:-trigger-image"
            class="flex-1 outline-none mt-0"
            data-orientation="horizontal"
            data-slot="tabs-content"
            data-state="active"
            id="radix-:r6:-content-image"
            role="tabpanel"
            style="animation-duration: 0s;"
            tabindex="0"
          >
            <div
              class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
            />
          </div>
          <div
            aria-labelledby="radix-:r6:-trigger-audio"
            class="flex-1 outline-none mt-0"
            data-orientation="horizontal"
            data-slot="tabs-content"
            data-state="inactive"
            hidden=""
            id="radix-:r6:-content-audio"
            role="tabpanel"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
